### ISSUE TITLE
Prototype task code refactor

### ISSUE THREAD

**Issue by msbtterswrth** [department-of-veterans-affairs/tmf-auth-exp-design-patterns#98]
This was broken out from #94.
### Background
As an engineer I would like to cleanup and refactor the 3 tasks that we have bootstrapped for the prefill pattern. We should try and really keep PRs below 500 LOC so either break this task into multiple PRs, or create more tasks if the work is extensive.

Some things to investigate:

- Repeated code should be consolidated, reusable common components should be utilized, and file organization should be optimized.
- Mock auth and staging environment workarounds. What can we do to make this better? The codespaces work as expected, but we would really like to have a way to get staging to cooperate for our patterns.
- ~~Could we point just our app in staging to use the Codespaces api url? That way the auth could be mocked and requests would be sent to that location instead of the staging api?~~ Due to the way that the api endpoints are configured in the frontend there is no way to really override the staging api endpoint setting with a different endpoint. The environment uses a 'frozen' object for just this reason that it is not able to be changed. Its good for security on production, but makes this idea not viable.

Acceptance Criteria
- [x] PR to isolate pattern code has been created and merged
- [x]  Existing pattern is available and a branch is set up to use in codespaces along with a backup branch so that this work can be merged into main and not affect the research prototype.

**Comment by adamwhitlock1** [comment]
Made comments on the main task here: https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/94#issuecomment-2311465215

PR draft was created, and waiting on research session to be complete for our work to continue on this, since there will be url updates that we don't want to push up right away.

**Comment by adamwhitlock1** [comment]
@msbtterswrth and @beckyphung 

I'd like to propose that we unblock this ticket by using an alternative branch of main for the research sessions and be able to merge things like this work into main so that we can continue progress on the engineering team.

I updated [my PR](https://github.com/department-of-veterans-affairs/vets-website/pull/31584) to include some details about how we can use a different branch of code for the research sessions enabling us to: not have to change the url, be able to do hotfixes and changes to the research session code if needed, and be able to have a backup of the state of the code used for research sessions just in case.

It boils down to a few key things:
- _any_ branch on vets-website can be used as the source of truth for the Codespace environment. That means we can change our Codespace to use any branch to run on our live url and this provides greater flexibility.
- I have created a `research-1` and `research-1-backup` branch so that we can use those branches for research, and switch to the main branch once research is finished. 
- By using `research-1` as our branch for the codespace, whatever changes are made to `main` will NOT impact user research sessions. The separation ensures that the user research environment remains stable and will not be affected by changes in the main codebase whatsoever.
- Since we have no sessions today I have switched the Codespace to use the `research-1` branch as an experiment and have verified that it builds and works as expected.
  
Does anyone have any objections to moving forward with our work to refactor the codebase and use the research-1 branch as the way to run our research sessions?

@bellepx0 and I have discussed this as well and agree that this would be an acceptable approach and has minimal risks.

**Comment by msbtterswrth** [comment]
this sounds like an excellent idea, and is inline wit general best practices for feature branching. I would propose a slight change to the branch title to be `research-mo-year` instead of 1,2,3 etc so this first one would be `research-08-2024`. Thoughts?

**Comment by adamwhitlock1** [comment]
sure, I can do that. Branches are free to make 😉
